app/services/processors/blog_posts_updater.rb

```diff
@@ -0,0 +1,25 @@
+module Processors
+  class BlogPostsUpdater
+    def call
+      wordpress_service.blog_posts.each do |blog_post|
+        BlogPost.create_with(
+          published_at: blog_post[:date],
+          slug: blog_post[:slug],
+          status: blog_post[:status],
+          url: blog_post[:URL],
+          technology: categorizer.technology_for(blog_post)
+        ).find_or_create_by!(blog_id: blog_post[:ID])
```

Comment:
What about using only `find_or_create_by` with a block?
```
BlogPost.find_or_create_by!(blog_id: blog_post[:ID]) do |blog_post|
  blog_post.published_at = blog_post[:date],
  blog_post.slug = blog_post[:slug],
  blog_post.status = blog_post[:status],
  blog_post. url = blog_post[:URL],
  blog_post.technology = categorizer.technology_for(blog_post)
end
```
Just a suggestion, i don't really know which one is better.

Reply:
I don't either, but I like yours best!